src/main/java/it/gov/pagopa/cgn/portal/service/BucketService.java

```diff
@@ -168,7 +168,8 @@ public void performBucketLoad(Long discountId) {
 
     @Transactional(Transactional.TxType.REQUIRES_NEW)
     public void deleteBucketCodes(Long discountId) {
-        discountBucketCodeRepository.deleteByDiscountId(discountId);
+        if (discountId != null)
```

Comment:
Do we have this scenario? just saying, in this case we have discount bucket codes that not refers to any discount right?

Reply:
This scenario happens only during the creation of "non-bucket" discounts. 
We perform the validation before having the discount id and validation is common for creation and update (as it is implemented now) and for "non-bucket" discount it cleans fields and eventually deletes any bucket code if present. We need to skip this deletion during creation, the most simple way is checking for null discountId.
Another way could be refactoring to have separated creation and update validations.